lib/graphql/resolvers/spreadsheet.resolver.ts

```diff
@@ -0,0 +1,52 @@
+import { Query } from "type-graphql";
+import { injectable } from "tsyringe";
+import { readSpreadsheet } from "../utilities/spreadsheet/setup";
+import { isEqual } from "lodash";
+import SpreadsheetService from "../services/Spreadsheet.service";
+import { SpreadsheetOverviewRevenueType, SpreadsheetOverviewDivisionsType } from "../schemas/Spreadsheet";
+
+@injectable()
+export default class spreadsheetResolver { 
+    constructor( private spreadsheetService : SpreadsheetService ) {} 
+
+    @Query( () => [SpreadsheetOverviewRevenueType] )
+    async getSpreadsheetOverviewRevenueData() {
+
+        const revenueData = await readSpreadsheet( ["Overview!B2:B", "Overview!D2:D", "Overview!F2:F" ] ); 
+        
+        if (!revenueData) return [] 
+        
+        const propertyNames = [ ['itemName', 'string'], ['budgetAmount', 'float'], ['notes', 'string']]
+        const startIndex = 1 ;
+        const endIndex =  revenueData?.at(0)?.values?.findIndex( obj => isEqual( obj, ['Sub Category'] ) )   || -1 
+
+        // Sanity check in case google sheet column is no longer called "Sub category"
+        if (endIndex === -1 ) throw new Error("Cannot find end index of revenue section in Overview sheet- looking for cell 'Sub Category' " );
```

Comment:
Thoughts on just returning empty array and then silently logging the error out instead of explicitly throwing the error?

Reply:
that sounds like a good idea to me